ding_spt_statistics.module

```diff
@@ -3,3 +3,153 @@
  * @file
  * Implementation of hooks and basic functionality goes here.
  */
+
+/**
+ * Implements hook_menu().
+ */
+function ding_spt_statistics_menu() {
+  $items = array();
+
+  $items['spt/redirect'] = array(
+    'title' => 'Ding SPT redirect',
+    'description' => 'Count click and redirect to the external resource',
+    'type' => MENU_CALLBACK,
+    'access arguments' => array('access content'),
+    'page callback' => 'ding_spt_statistics_redirect',
+  );
+  $items['spt/statistics'] = array(
+    'title' => 'Ding SPT statistics',
+    'description' => 'Count click and redirect to the external resource',
+    'type' => MENU_CALLBACK,
+    'access arguments' => array('view spt statistics'),
+    'page callback' => 'ding_spt_statistics_display',
+  );
+
+  return $items;
+}
+
+/**
+ * Implements hook_permission().
+ */
+function ding_spt_statistics_permission() {
+  return array(
+    'view spt statistics' => array(
+      'title' => t('View SPT statistics'),
+      'description' => t('View statistics for clicked external SPT resources.'),
+    ),
+  );
+}
+
+/**
+ * Count click and redirect to the resource.
+ */
+function ding_spt_statistics_redirect() {
+  // Register statistical data.
+  $hostname = !empty($_GET['hostname']) ? $_GET['hostname'] : FALSE;
+  ding_spt_statistics_register($hostname);
+
+  // Redirect to the external resource.
+  $path = !empty($_GET['path']) ? $_GET['path'] : FALSE;
+  drupal_goto($path);
+}
+
+/**
+ * Register hostname into the statistics table.
+ *
+ * @param $hostname string
+ *   Name of the external host.
+ */
+function ding_spt_statistics_register($hostname) {
+  if (!$hostname) {
+    return;
+  }
+  $result = db_select('ding_spt_statistics', 'dss')
+    ->fields('dss', array('hid', 'count'))
+    ->condition('hostname', $hostname)
+    ->execute()
+    ->fetchAssoc()
+  ;
+
+  // Update table statistics with new values.
+  if (!$result) {
+    $result['hid'] = db_insert('ding_spt_statistics')
+      ->fields(array(
+        'hostname' => $hostname,
+        'count' => 1,
+      ))
+      ->execute();
+  }
+  else {
+    db_update('ding_spt_statistics')
+      ->expression('count', 'count + 1')
+      ->condition('hostname', $hostname)
+      ->execute();
+  }
+
+  // Insert new record to the data table.
+  db_insert('ding_spt_hits')
+    ->fields(array(
+      'hid' => $result['hid'],
+      'accessed' => REQUEST_TIME,
+    ))
+    ->execute();
+}
+
+/**
+ * Display statistics of the hits as a table.
+ *
+ * @return string
+ *   Themed table.
+ */
+function ding_spt_statistics_display() {
+  // Get data from db.
+  $query = db_select('ding_spt_statistics', 's');
+  $query->leftJoin('ding_spt_hits', 'h', 's.hid = h.hid');
+  $query->fields('s', array('hostname', 'count'));
+
+  $query->addExpression('SUM(CASE WHEN unix_timestamp(now()) - h.accessed < 86400 THEN 1 ELSE 0 END)', 'day');
```

Comment:
please use one notation - if you started to write with uppercase, then keep it
 unix_timestamp(now()) =>to uppercase